### DISCUSSION TITLE
Consistency handling `null` values in `MultiSelect List` and `MultiSelect Hash`

### DISCUSSION THREAD

**Discussion by glenveegee** [jmespath-community/jmespath.spec#156] [Q&A] [answered]
Under the assumption, the jmespath playground at jmespath.org represents the canonical output for any expression,  there appear to be some interesting inconsistencies. Either these are undocumented in the spec, or the spec isn't clear on how to handle these edge cases, or there are language-specific implementations that treat the following edge cases differently.

Consider the following... what would you expect

1. Null literal piped to multiselect list

```javascript
// expression
`null` | [@]

// Should it be : [null]
// OR                : null
```

2. Null literal piped to multiselect hash

```javascript
// expression
`null` | {foo: @}

// Should it be : {"foo": null}
// OR                : null
```

**Comment by springcomp** [comment, marked as answer]
Hi @glenveegee We have answered this question and [standardized this behaviour at JMESPath Community](https://github.com/jmespath-community/jmespath.spec/blob/main/jep-019-pipe-expressions-evaluation.md).

We believe your second example should return `` {"foo": null} ``.

You are right that this is not part of the compliance tests and some implementations may still be "broken".
